### ISSUE TITLE
Cores should be updated when a new version is installed (Windows)

### ISSUE THREAD

**Issue by ppvnf** [libretro/RetroArch#15323]
Unless I am mistaken, in the Google Play version the cores are updated every time a new version is installed. Would it be possible to implement this feature on Windows as well? I often forget to update the cores and Retroarch version updates are easier to track. Thank you.

**Comment by hizzlekizzle** [comment]
You could download the stable build core snapshot and place them all in your cores directory overwriting the conflicts when you install.

There is at least one reason this would be problematic, which is that core updates can break savestates, so forcibly updating all of the cores could invalidate users' progress.

**Comment by ppvnf** [comment]
> You could download the stable build core snapshot and place them all in your cores directory overwriting the conflicts when you install.
I think that would work, however I still think it should be built into the installer (but as an option) to make it easier.

> There is at least one reason this would be problematic, which is that core updates can break savestates, so forcibly updating all of the cores could invalidate users' progress.
I didn't know about that (does the regular update warns about it?). How about making it a checkbox during the install of a new version?

**Comment by keithbowes** [comment]
I disagree.  Often updates break, as evidenced by all the commits that merely revert earlier commits.  I agree about stable versions of cores, but AFAIK, there's still no system to identify stable versions.  Asking users if they want to update their cores is one thing, but potentially breaking users' setups without their consent is something else.

**Comment by odditude42** [comment]
if this were to be added (which I agree is inadvisable), I suggest adding it as an option with default value DISABLED and a warning message that it can lead to broken save states.

**Comment by ppvnf** [comment]
how about prompting the core update when RetroArch is opened or when a core/content is loaded? (I agree that a warning about breaking save states should be shown)

**Comment by RobLoach** [comment]
A message prompting the user would get VERY annoying.

libretro cores are version independent from the version of RetroArch. Meaning it doesn't matter what version of RetroArch you use to what core you're on. They are all targeting libretro API v1. If you're concerned about updating the cores, then use the "Update Installed Cores" option in the Online Updater.
